crates/driver/src/domain/competition/solution/trade.rs

```diff
@@ -13,14 +20,72 @@ pub enum Trade {
 pub struct Fulfillment {
     pub order: competition::Order,
     /// The amount executed by this fulfillment. See
-    /// [`competition::order::Partial`]. If the order is not partial, the target
-    /// amount must equal the amount from the order.
+    /// [`competition::order::Partial`]. If the order is not partial, the
+    /// executed amount must equal the amount from the order.
     pub executed: competition::order::TargetAmount,
 }
 
+impl Fulfillment {
+    /// Calculate the settlement contract input and output amounts executed by
+    /// this trade.
+    pub fn execution(&self, clearing_prices: &ClearingPrices) -> Result<Execution, Error> {
+        let input = self.executed.to_asset(&self.order);
+        let (input_price, output_price) = {
```

Comment:
I think in this context it makes more sense to think in terms of `sell_amount` and `buy_amount`. Where `sell_amount` always flows into the contract and `buy_amount` always flows out of the contract. Using these names makes the computation easier to follow IMO.